### PR TITLE
Set distiluse model 4 good

### PR DESCRIPTION
**Pull request by psorianom**
## Reference to a related issue

Fixes https://github.com/etalab-ia/piaf-ml/issues/85

## Why is the change needed

Long time ago, we were using the sentence-transformers `distiluse` (dim-512) model without any hassle. One day, we decided to keep tabs on what version of the model we were using, for reproducibility sake. At the time, haystack did not have the option to use a specific version of a `sentence-transformers` model, so we changed to a transformers retriever with the `distilbert` (dim-768) model, because `distiluse` was not available in HF model's hub.

We were having a mix of `distiluse`, `distilbert` and a little bit of a mess with the dimensions in our code.

Also, the performance was not as good as with `distiluse`. Lately, `distiluse`has been made available in HF model's hub, so it is easily instantiable as a `transformers` model. This time the model has (dim-768) though :shrug:. 

## Description of the change

I removed all the sentence-transformers `distiluse` references and replaced them with the transformers' `sentence-transformers/distiluse-base-multilingual-cased-v2` identifier. All configurations mentioning `512` as number of dimensions were changed to `768`.

### CODE REVIEW

File: test/test_eval_retriever.py [hunk 1]
```diff
@@ -126,7 +126,8 @@ def test_prepare_fquad_eval(document_store):
 @pytest.mark.parametrize("retriever_type,recall_expected,mrr_expected",
                          [
                              ("bm25", 15 / 16, 14 / 16),
-                             ("sbert", 1., ((4 * 1 / 2) + (1 * 1/3) + 11) / 16),
+                             ("sbert", 1., (3 * 1 / 2 + 13) / 16),
+                             # ("sbert", 1, (5 * 1 / 2 + 11) / 16)]),
```
Comment (Rob192) [inline test/test_eval_retriever.py:130]: shall we remove the comment ?

File: test/test_eval_retriever.py [hunk 2]
```diff
@@ -126,7 +126,8 @@ def test_prepare_fquad_eval(document_store):
 @pytest.mark.parametrize("retriever_type,recall_expected,mrr_expected",
                          [
                              ("bm25", 15 / 16, 14 / 16),
-                             ("sbert", 1., ((4 * 1 / 2) + (1 * 1/3) + 11) / 16),
+                             ("sbert", 1., (3 * 1 / 2 + 13) / 16),
```
Comment (Rob192) [inline test/test_eval_retriever.py:129]: Looks better

**Review by Rob192 — CHANGES_REQUESTED** [review]
We also need to update the client pipelines.yaml files to reflect this change

**Review by guillaumecherel — APPROVED** [review]

File: test/test_eval_retriever.py (hunk 1 above)
Comment (psorianom) [reply test/test_eval_retriever.py:130]: yep!

**Comment by psorianom** [comment]
> We also need to update the client pipelines.yaml files to reflect this change

ah yes, I forgot those. Will do.

**Comment by guillim** [comment]
Question : should we update this **"dims"** value as well to **728** ? Not sure if it's overridden later in the application
https://github.com/etalab-ia/piaf-ml/blob/8c71d614353a34a0762c2da6cc1080de8f2c5cc2/deployment/roles/prepare_data/templates/insert_data.py#L94

**Comment by psorianom** [comment]
> Question : should we update this **"dims"** value as well to **728** ? Not sure if it's overridden later in the application
> 
> https://github.com/etalab-ia/piaf-ml/blob/8c71d614353a34a0762c2da6cc1080de8f2c5cc2/deployment/roles/prepare_data/templates/insert_data.py#L94

Nope, you're right, we should only have 768. Nothing uses 512 dimensions any more, for now.

**Review by Rob192 — APPROVED** [review]

**Pull request merged**